src/Manager.cc

```diff
@@ -1106,35 +1111,50 @@ void ManagerPrivate::LoadPlugin(const tinyxml2::XMLElement *_elem)
 
   if (pathToLib.empty())
   {
-    ignerr << "Failed to find the path to library[" << file << "]. "
-      << "Try adding the path to the IGN_LAUNCH_PLUGIN_PATH environment "
+    // TODO(CH3): Deprecated. Remove on ticktock.
+    // This tries to find one more time with IGN_LAUNCH_PLUGIN_PATH instead of
+    // GZ_LAUNCH_PLUGIN_PATH
+    systemPaths.SetPluginPathEnv("IGN_LAUNCH_PLUGIN_PATH");
+    pathToLib = systemPaths.FindSharedLibrary(file);
+  }
+
+  if (pathToLib.empty())
+  {
+    gzerr << "Failed to find the path to library[" << file << "]. "
+      << "Try adding the path to the GZ_LAUNCH_PLUGIN_PATH environment "
       << "variable.\n";
     return;
   }
+  else
```

Comment:
I think all successful code paths will get here, even if the plugin was found in `~/.gz` or `GZ_LAUNCH_PLUGIN_PATH`. I think this block needs to be moved into the `IGN_LAUNCH_PLUGIN_PATH` above. And I'd move this line into that block too, so we catch all ignition paths:

```
  systemPaths.AddPluginPaths(
    gz::common::joinPaths(homePath, ".ignition", "gazebo", "plugins"));
```

Reply:
Whoops! Brainfart

I was trying to be smart and avoid another condition check, but I'm just going to bite the bullet here

https://github.com/gazebosim/gz-launch/pull/163/commits/bb117b30b549b679f05c1020ca5a816c2f277f61

Reply:
I'm not sure if moving the line helps, if we move the line, then systemPaths won't have `~/.ignition/gazebo/plugins` available when it's using `GZ_LAUNCH_PLUGIN_PATH`.

`SetPluginPathEnv()` shouldn't erase the paths added by `AddPluginPaths`, unless I got that wrong?

Reply:
I force-pushed and updated accordingly: https://github.com/gazebosim/gz-launch/pull/163/commits/a2c3ecf83f9135a75d562f10a642bfcbe023db2c

```
  std::string pathToLib;
  if (common::exists(file))
    pathToLib = file;
  else
    pathToLib = systemPaths.FindSharedLibrary(file);

  if (pathToLib.empty())
  {
    // TODO(CH3): Deprecated. Remove on ticktock.
    // This tries to find one more time with IGN_LAUNCH_PLUGIN_PATH instead of
    // GZ_LAUNCH_PLUGIN_PATH
    systemPaths.SetPluginPathEnv("IGN_LAUNCH_PLUGIN_PATH");
    pathToLib = systemPaths.FindSharedLibrary(file);

    if (!pathToLib.empty())
    {
      gzwarn << "Found plugin [" << pathToLib
      << "] using deprecated environment variable [IGN_LAUNCH_PLUGIN_PATH]."
         " Please use [GZ_LAUNCH_PLUGIN_PATH] instead." << std::endl;
    }
  }

  if (pathToLib.empty())
  {
    gzerr << "Failed to find the path to library[" << file << "]. "
      << "Try adding the path to the GZ_LAUNCH_PLUGIN_PATH environment "
      << "variable.\n";
    return;
  }
```

Reply:
Oh, I think it was better how it was before, with `~/.ignition/gazebo/plugins` inside the if.

> then systemPaths won't have ~/.ignition/gazebo/plugins available when it's using GZ_LAUNCH_PLUGIN_PATH.

That's ok, they're all alternative lookup paths. The plugin will either be in pne of the paths defined in `GZ_.._PATH` or in `~/.ignition/gazebo/plugins`. They don't need to be used together.

`GZ_LAUNCH_PLUGIN_PATH=/tmp/plugins:/home/banana/plugins`

> SetPluginPathEnv() shouldn't erase the paths added by AddPluginPaths

Nope, it doesn't erase them. So here's the idea:

1. Tell it to look into the paths from `GZ_LAUNCH_PLUGIN_PATH` and also in `~/.gz/...`
2. If it's not found, look into the deprecated paths, from `IGN_LAUNCH_PLUGIN_PATH` and `~/.ignition/...`
3. If found with the deprecated paths, print warning

Reply:
Alright, will revert accordingly

Reply:
https://github.com/gazebosim/gz-launch/pull/163/commits/b51b2aa6c47b26303ec176c76d91594126b6669a